core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -444,6 +443,7 @@ private[spark] class TaskSchedulerImpl(
     // of locality levels so that it gets a chance to launch local tasks on all of them.
     // NOTE: the preferredLocality order: PROCESS_LOCAL, NODE_LOCAL, NO_PREF, RACK_LOCAL, ANY
     for (taskSet <- sortedTaskSets) {
+      val availableSlots = availableCpus.map(c => c / CPUS_PER_TASK).sum
```

Comment:
note: in https://github.com/apache/spark/pull/25946/files#diff-d4000438827afe3a185ae75b24987a61R455 the `else` could become an `else if (availableSlots > 0)` to avoid iterating over the remaining task sets when all available slots have been taken already, but that's a micro-optimization orthogonal to the fix.

Reply:
to avoid reiterating over the availableCpus array, we could also have one var that gets passed and updated inside resourceOfferSingleTaskSet, but that would likely be an overoptimization of this O(#num_offers) sum.